source/MaterialXGenShader/ShaderGenerator.h

```diff
@@ -147,6 +147,13 @@ class ShaderGenerator
         return nullptr;
     }
 
+    /// Get a list of implementations used for shader generation
+    const std::unordered_map<string, ShaderNodeImplPtr>& getImplementationsUsed() const
```

Comment:
It is quite hard to tell what implementations are used by shader gen externally so allow access to implementation list directly here.

Reply:
I wanted to add as little overhead as possible but adding this does help hide the internals so I can do the change.

Reply:
Overhead / complexity should be the same, as far as I could tell, since you already called count() on the returned maps in the library loop below. count() and find() has the same complexity.